synthnn/core/musical_synthesis.py

```
@@ -426,13 +426,15 @@ class MusicalResonantNetwork(ResonantNetwork):
     Musical extension of ResonantNetwork with synthesis capabilities.
     """
     
-    def __init__(self, name: str = "musical_network", 
+    def __init__(self, name: str = "musical_network",
                  mode: str = "ionian",
-                 base_freq: float = 440.0):
+                 base_freq: float = 440.0,
+                 mode_detector: Optional[Any] = None):
```

Ruff (0.11.9)

Dynamically typed expressions (typing.Any) are disallowed in `mode_detector`

(ANN401)